content/en/guide/v10/forms.md

```diff
@@ -117,61 +176,241 @@ class MySelect extends Component {
+      <form onSubmit={onSubmit}>
+        <label>
+          First Name:{' '}
+          <input name="firstName" />
+        </label>
+        <label>
+          Last Name:{' '}
+          <input name="lastName" />
+        </label>
+        <button>Submit</button>
+      </form>
+      {fullName && <p>Hello {fullName}</p>}
+    </div>
+  );
+}
+
+// --repl-after
+render(<FullNameForm />, document.getElementById("app"));
+```
+
+</tab-group>
+
+> **Note**: Whilst it's quite common to see React & Preact forms that link every input field to component state, it's often unnecessary and can get unwieldy. As a very loose rule of thumb, you should prefer using `onSubmit` and the [`FormData`](https://developer.mozilla.org/en-US/docs/Web/API/FormData) API in most cases, using component state only when you need to. This reduces the complexity of your components and may skip unnecessary rerenders.
+
+## Controlled & Uncontrolled Components
+
+When talking about form controls you may encounter the terms "Controlled Component" and "Uncontrolled Component". These terms refer to whether or not the form control value is explicitly managed by the component. Generally, you should try to use _Uncontrolled_ Components whenever possible, the DOM is fully capable of handling `<input>`'s state:
+
+```jsx
+// Uncontrolled, because Preact doesn't set the value
+<input onInput={myEventHandler} />;
+```
+
+However, there are situations in which you might need to exert tighter control over the input value, in which case, _Controlled_ Components can be used.
+
+```jsx
+// Controlled, because Preact sets the value
+<input value={myValue} onInput={myEventHandler} />;
+```
+
+Preact has a known issue with controlled components: rerenders are required for Preact to exert control over input values. This means that if your event handler doesn't update state or trigger a rerender in some fashion, the input value will not be controlled, sometimes becoming out-of-sync with component state.
+
+An example of one of these problematic situations is as such: say you have an input field that should be limited to 3 characters. You may have an event handler like the following:
+
+```js
+const onInput = (e) => {
+  if (e.currentTarget.value.length <= 3) {
+    setValue(e.currentTarget.value);
+  }
+}
+```
+
+The problem with this is in the cases where the input fails that condition: because we don't run `setValue`, the component doesn't rerender, and because the component doesn't rerender, the input value is not correctly controlled. However, even if we did add a `else { setValue(value) }` to that handler, Preact is smart enough to detect when the value hasn't changed and so it will not rerender the component. This leaves us with [`refs`](/guide/v10/refs) to bridge the gap between the DOM state and Preact's state.
+
+> For more information on controlled components in Preact, see [Controlled Inputs](https://www.jovidecroock.com/blog/controlled-inputs) by Jovi De Croock.
```

Comment:
Tried to expound more on the controlled input issue, rather than jumping straight into "use refs", which I wrote before. I hope it's better? Basically trying to condense & transplant Jovi's writing.

Feedback on the wording here very much appreciated, as this is a bit of a prickly issue that users may run into.